### ISSUE TITLE
ViT INT8 QAT Accuracy Degradation

### ISSUE THREAD

**Issue by lingffff** [NVIDIA/FasterTransformer#267]
### Description

```shell
Branch: up to date version
GPU: 8 x Tesla V100 32G
```

### Reproduced Steps

```shell
1. Follow the steps of docs/vit_guide.md
2. ......
3. cd examples/pytorch/vit/ViT-quantization
4. edit qat.sh: --nproc_per_node 1 to 8
5. bash qat.sh, and only got 81.888% max top1 accuracy

In the end of examples/pytorch/vit/ViT-quantization/README.md:

### Improvement brought by QAT (under mode=2)
As shown below, the accuracy gap is narrowed down from 2.15% to 0.65%.
|name| resolution|Original Accuracy|PTQ(mode=2)|QAT(mode=2)|
|:---:|:---:|:---:|:---:|:---:|
|ViT-B_16|384x384|83.97%|81.82%(-2.15%)|83.32(-0.65%)|

QAT acc is 83.32%, but I only got 81.888%. 
What may cause this degradation?
```

**Comment by Njuapp** [comment]
I can get the result by the following hyper-parameters @lingffff 
> python -m torch.distributed.launch --nproc_per_node 4 \
--master_port 12345 train.py \
--train \
--name vit \
--data-path /mount \
--model_type ViT-B_16 \
--pretrained_dir /mount/checkpoint/ViT-B_16_ft2_99.99_81.948.pth  \
--distill --teacher /mount/checkpoint/ViT-B_16.npz \
--quant-mode ft2 \
--percentile 99.99 \
--batch-size 8 \
--qat-lr 1e-3

**Comment by lingffff** [comment]
> I can get the result (top1 acc 83.28%) by the following hyper-parameters @lingffff
> 
> > python -m torch.distributed.launch --nproc_per_node 4 
> > --master_port 12345 train.py 
> > --train 
> > --name vit 
> > --data-path /mount 
> > --model_type ViT-B_16 
> > --pretrained_dir /mount/checkpoint/ViT-B_16_ft2_99.99_81.948.pth  
> > --distill --teacher /mount/checkpoint/ViT-B_16.npz 
> > --quant-mode ft2 
> > --percentile 99.99 
> > --batch-size 8 
> > --qat-lr 1e-3

Thanks!
What about the --num-epochs ?

**Comment by Njuapp** [comment]
> > I can get the result (top1 acc 83.28%) by the following hyper-parameters @lingffff
> > > python -m torch.distributed.launch --nproc_per_node 4
> > > --master_port 12345 train.py
> > > --train
> > > --name vit
> > > --data-path /mount
> > > --model_type ViT-B_16
> > > --pretrained_dir /mount/checkpoint/ViT-B_16_ft2_99.99_81.948.pth
> > > --distill --teacher /mount/checkpoint/ViT-B_16.npz
> > > --quant-mode ft2
> > > --percentile 99.99
> > > --batch-size 8
> > > --qat-lr 1e-3
> 
> Thanks! What about the --num-epochs ?

I used the default parameter, num_epochs=10, but the model with best accuracy is obtained at the 7-th epoch.

**Comment by lingffff** [comment]
> > > I can get the result (top1 acc 83.28%) by the following hyper-parameters @lingffff
> > > > python -m torch.distributed.launch --nproc_per_node 4
> > > > --master_port 12345 train.py
> > > > --train
> > > > --name vit
> > > > --data-path /mount
> > > > --model_type ViT-B_16
> > > > --pretrained_dir /mount/checkpoint/ViT-B_16_ft2_99.99_81.948.pth
> > > > --distill --teacher /mount/checkpoint/ViT-B_16.npz
> > > > --quant-mode ft2
> > > > --percentile 99.99
> > > > --batch-size 8
> > > > --qat-lr 1e-3
> > 
> > 
> > Thanks! What about the --num-epochs ?
> 
> I used the default parameter, num_epochs=10, but the model with best accuracy is obtained at the 7-th epoch.

OK I will try. Thanks again.

**Comment by lingffff** [comment]
I now get 83.104, looks good.
